### ISSUE TITLE
Dockerfile inappropriate "as" highlighting

### ISSUE THREAD

**Issue by cjone0102** [neovim/neovim#11862]
- `nvim --version`: 
```
❯ nvim --version
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/config -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/src -I/usr/local/include -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/deps-build/include -I/usr/local/opt/gettext/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/src/nvim/auto -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/include
Compiled by brew@Catalina.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.4.3/share/nvim"

Run :checkhealth for more info
```

- `vim -u DEFAULTS` (version: ) behaves differently?
```
❯ vim -u DEFAULTS --version
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/config -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/src -I/usr/local/include -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/deps-build/include -I/usr/local/opt/gettext/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/src/nvim/auto -I/tmp/neovim-20191107-85913-1wbgqz6/neovim-0.4.3/build/include
Compiled by brew@Catalina.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.4.3/share/nvim"

Run :checkhealth for more info
```

- Operating system/version: MacOS Catalina
- Terminal name/version: iTerm2 Build 3.3.8beta1, zsh 5.7.1 (x86_64-apple-darwin16.7.0)
- `$TERM`: 
```
❯ echo "$TERM"
screen-256color
```

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
<Esc>:set filetype=Dockerfile<CR>
```

Enter in the buffer `FROM ubuntu as base`

### Actual behaviour
The word "as" (between ubuntu and base) is highlighted.  The letters "as" in the word "base" are also highlighted.

### Expected behaviour
The word as (between ubuntu and base) is highlighted.  The letters "as" **should not** be highlighted.

**Comment by cjone0102** [comment]
It's easier to see the problem with a colorful theme.  Here it is with my normal color scheme: 
![Window capture 3](https://user-images.githubusercontent.com/60937477/74372384-8cc84100-4da8-11ea-917b-ab1d4aa72c4c.png)

**Comment by teto** [comment]
does vim behave differently ? if no it means we are late a patch https://github.com/vim/vim/blob/master/CONTRIBUTING.md#syntax-indent-and-other-runtime-files  . Seems like it was modified recently:
in vim 560979ed4f0216f902a2c247e937f00a27dcb198

**Comment by magikid** [comment]
Ah, yeah, looks like vim was just updated to fix this on 27 Jan: https://github.com/vim/vim/blob/560979ed4f0216f902a2c247e937f00a27dcb198/runtime/syntax/dockerfile.vim

**Comment by rafikdraoui** [comment]
There has been many improvements to `syntax/dockerfile.vim` upstream since the last sync a year ago: https://github.com/vim/vim/blob/master/runtime/syntax/dockerfile.vim

**Comment by jarimayenburg** [comment]
Improvements from last january/february have been merged with #13404!

**Comment by mhinz** [comment]
This is fixed on master. (Probably due to https://github.com/neovim/neovim/pull/14424)
